src/eei.cpp

```diff
@@ -199,6 +199,9 @@ string toHex(evmc_uint256be const& value) {
 
       takeGas(gas);
 
+      heraAssert((ffsl(gas) + ffsl(memory.size()) <= 64), "Memory gas calculation overflow."); //may need to find alternative to ffsl for cross-libc portability
+      takeGas(gas * memory.size() / GasSchedule::memoryPageSize * GasSchedule::memoryCostPerPage + 1); //round gas cost up
```

Comment:
Unsure whether this should be rounded up or not.

Reply:
Not sure. What are the pros/cons for each?